pilot/pkg/security/trustdomain/bundle.go

```diff
@@ -90,10 +90,15 @@ func (t Bundle) replaceTrustDomains(principal, trustDomainFromPrincipal string)
 		// as-is for the matched trust domain. For others, replace the trust domain with the new trust domain
 		// or alias.
 		var newPrincipal string
+		var err error
 		if suffixMatch(td, trustDomainFromPrincipal) {
 			newPrincipal = principal
 		} else {
-			newPrincipal = replaceTrustDomainInPrincipal(td, principal)
+			newPrincipal, err = replaceTrustDomainInPrincipal(td, principal)
+			if err != nil {
+				authzLog.Errorf("Failed to replace trust domain with %s from principal %s: %v", td, principal, err)
```

Comment:
The code here only log the error and continue. Should the error be propagated to the caller?

Reply:
IMHO, I think it would be OK to just log and continue here. Since even if we propagate the error to its (one-level) upper layer, we would probably do the same logging and continuing, similar to `getTrustDomainFromSpiffeIdentity`. WDYT?